crates/core/src/sql/execute.rs

```diff
@@ -21,17 +28,27 @@ pub struct StmtResult {
 // TODO(cloutiertyler): we could do this the swift parsing way in which
 // we always generate a plan, but it may contain errors
 
-pub(crate) fn collect_result(result: &mut Vec<MemTable>, r: CodeResult) -> Result<(), DBError> {
+pub(crate) fn collect_result(result: &mut Vec<MemTable>, updates: &mut Vec<DatabaseTableUpdate>, r: CodeResult) -> Result<(), DBError> {
     match r {
         CodeResult::Value(_) => {}
         CodeResult::Table(x) => result.push(x),
         CodeResult::Block(lines) => {
             for x in lines {
-                collect_result(result, x)?;
+                collect_result(result, updates, x)?;
             }
         }
         CodeResult::Halt(err) => return Err(DBError::VmUser(err)),
-        CodeResult::Pass => {}
+        CodeResult::Pass(x) => match x {
+            None => {},
+            Some(update) => {
+                updates.push(DatabaseTableUpdate {
+                    table_name: update.table_name,
+                    table_id: update.table_id,
+                    inserts: update.inserts.into(),
+                    deletes: update.deletes.into()
+                });
```

Comment:
That's the idea, except the types are slightly different. Where would you place DatabaseTableUpdate so its visible to both crates, and it can still impl its traits?

Or is it preferred to add spacetimedb-core as a vm dependency?